spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskMetrics.java

```diff
@@ -88,41 +92,47 @@
 	 */
 	public static final String TASK_EXCEPTION_TAG = "task.exception";
 
-	private Timer.Sample taskSample;
-
 	private LongTaskTimer.Sample longTaskSample;
 
-	private Throwable exception;
+	private Observation observation;
 
 	public void onTaskStartup(TaskExecution taskExecution) {
-		LongTaskTimer longTaskTimer = LongTaskTimer
+		if (this.registry != null) {
+			this.observation = Observation.start(SPRING_CLOUD_TASK_METER, this.registry)
+				.lowCardinalityTag(TASK_EXIT_CODE_TAG, String.valueOf(taskExecution.getExitCode()))
```

Comment:
We were supposed to migrate this to using `TagsProvider` approach - check https://github.com/spring-projects/spring-batch/pull/4065/files for reference

Reply:
@marcingrzejszczak @jonatan-ivanov I updated the PR based on your code review comments.   First off thanks for the review!    

One note:  Since task does not have any High Cardinality tags and one is required, a `TagKey` for  HighCardinality tags was added.  It only contains a task.name tag.